components/playlist/browser/playlist_service.cc

```diff
@@ -747,6 +798,41 @@ void PlaylistService::RemoveLocalDataForItem(
   }
 }
 
+void PlaylistService::OnMediaFileDownloadFinished(
+    bool update_media_src_and_retry_on_fail,
+    mojom::PlaylistItemPtr item,
+    const std::string& media_file_path) {
+  DCHECK(item);
+  DCHECK(IsValidPlaylistItem(item->id));
+
+  VLOG(2) << __func__ << ": " << item->id << " result path" << media_file_path;
+
+  if (media_file_path.empty() && update_media_src_and_retry_on_fail) {
+    VLOG(2) << __func__ << ": " << item->id
+            << " try updating media src and download";
+
+    base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+        FROM_HERE, base::BindOnce(&PlaylistService::RecoverLocalDataForItem,
```

Comment:
Is there any possibility that "download failed and recover" will be repeated indefinitely?

Reply:
Well spotted! I should count the number of retrial.

Reply:
Oh, it's not likely happen, as update_media_src_and_retry_on_fail will be false when update_media_src_before_recovery is true.

Reply:
Ok, great. Thanks for checking!